File: template/app/src/main/java/com/xmartlabs/template/ui/login/LoginFragment.java
package com.xmartlabs.template.ui.login;

import android.content.Intent;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;

import com.hannesdorfmann.fragmentargs.annotation.FragmentWithArgs;
import com.xmartlabs.template.R;
import com.xmartlabs.template.TemplateApplication;
import com.xmartlabs.template.ui.Henson;
import com.xmartlabs.template.ui.common.TemplateFragment;

import butterknife.BindView;
import butterknife.OnClick;

@FragmentWithArgs
public class LoginFragment extends TemplateFragment<LoginView, LoginPresenter> implements LoginView {
  @BindView(R.id.log_in_button)
  Button loginButton;
  @BindView(R.id.progress_bar)
  ProgressBar progressBar;

  @NonNull
  private final LoginPresenter presenter = LoginPresenter.builder().build();

  @NonNull
  @Override
  protected LoginPresenter createPresenter() {
    return presenter;
  }

  @LayoutRes
  @Override
  protected int getLayoutResId() {
    return R.layout.fragment_login;
  }

  @Override
  public void setIsLoading(boolean loading) {
    loginButton.setVisibility(loading ? View.GONE : View.VISIBLE);
    progressBar.setVisibility(loading ? View.VISIBLE : View.GONE);
  }

  @Override
  public void gotoRecyclerExampleActivity() {
    Intent intent = Henson.with(TemplateApplication.getContext())
        .gotoRecyclerExampleActivity()
        .build()
        .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

    startActivity(intent);
  }

  @OnClick(R.id.log_in_button)
  void onLoginButtonClicked() {
    presenter.loginButtonClicked();
  }
}

File: template/app/src/main/java/com/xmartlabs/template/ui/login/LoginFragment.kt
package com.xmartlabs.template.ui.login

import android.content.Intent
import android.view.View
import com.hannesdorfmann.fragmentargs.annotation.FragmentWithArgs
import com.xmartlabs.template.App
import com.xmartlabs.template.R
import com.xmartlabs.template.ui.Henson
import com.xmartlabs.template.ui.common.TemplateFragment
import kotlinx.android.synthetic.main.fragment_login.*
import javax.inject.Inject

@FragmentWithArgs
class LoginFragment : TemplateFragment<LoginView, LoginPresenter>(), LoginView {
  @Inject
  override lateinit var presenter: LoginPresenter

  override val layoutResId = R.layout.fragment_login

  override fun setup() {
    loginButton.setOnClickListener { presenter.loginButtonClicked() }
  }

  override fun setIsLoading(loading: Boolean) {
    loginButton.visibility = if (loading) View.GONE else View.VISIBLE
    progressBar.visibility = if (loading) View.VISIBLE else View.GONE
  }

  override fun gotoRecyclerExampleActivity() {
    val intent = Henson.with(App.context)
        .gotoRecyclerExampleActivity()
        .build()
        .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
    startActivity(intent)
  }
}
